CMakeLists.txt

```diff
@@ -96,8 +96,12 @@ if (WITH_CUDA)
     else()
       set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -O3")
     endif()
-    set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -gencode=arch=compute_20,code=sm_20 -gencode=arch=compute_30,code=sm_30 -gencode=arch=compute_50,code=sm_50")
-
+    set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -gencode=arch=compute_20,code=sm_20 \
+                                            -gencode=arch=compute_30,code=sm_30 \
+                                            -gencode=arch=compute_35,code=sm_35 \
+                                            -gencode=arch=compute_50,code=sm_50 \
+                                            -gencode=arch=compute_52,code=sm_52 \
+                                            -gencode=arch=compute_52,code=compute_52")
```

Comment:
My thoughts exactly. Doesn't this wreck compile time? Considering that cuda already takes forever for one arch.